test/parallel/test-assert-strict-typecheck.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// The following deepEqual tests might seem very weird.
+// They just describe what it is now.
+// That is why we discourage using deepEqual in our own tests.
+
+const date = new Date('2016');
+
+// See #10258
+function FakeDate() {}
+FakeDate.prototype = Date.prototype;
+const fake = new FakeDate();
+
+/* eslint-disable no-restricted-properties */
+// We are testing deepEqual!
```

Comment:
I think it's fine the way you did it, but if you want to get even more granular with the linting options, you can disable the check on just the lines that need it by either using `// eslint-disable-line no-restricted-properties` or `// eslint-disable-next-line no-restricted-properties` (depending on if you can fit the comment on the same line or if it needs to go on the line right before the `assert.deepEqual()`).